app/models/signup_state.rb

```diff
@@ -49,12 +49,36 @@ def trusted_email?
   end
 
   def trusted_external_uuid
-    trusted? ? trusted_data['external_user_uuid'] : nil
+    trusted? ? trusted_data['uuid'] : nil
   end
 
   def confirmed;  verified;  end
   def confirmed?; verified?; end
 
+  def linked_external_uuid
+    UserExternalUuid.find_by_uuid(trusted_external_uuid)
+  end
+
+  def suggested_username
+    if trusted_email? && LookupUsers.by_verified_email(trusted_data['email']).any?
+      trusted_data['email']
+    end
```

Comment:
This and the `on_user_authentication` code steer us away from skinny models.  So that this model doesn't have to know about the `LookupUsers` library method, can this code go elsewhere?  The view or somewhere.  A helper maybe.